src/libraries/System.Private.CoreLib/src/System/Int16.cs

```diff
@@ -1021,8 +1021,8 @@ public static bool TryCreate<TOther>(TOther value, out short result)
         /// <inheritdoc cref="IShiftOperators{TSelf, TResult}.op_RightShift(TSelf, int)" />
         static short IShiftOperators<short, short>.operator >>(short value, int shiftAmount) => (short)(value >> shiftAmount);
 
-        // /// <inheritdoc cref="IShiftOperators{TSelf, TResult}.op_UnsignedRightShift(TSelf, int)" />
-        // static short IShiftOperators<short, short>.operator >>>(short value, int shiftAmount) => (short)((ushort)value >> shiftAmount);
+        /// <inheritdoc cref="IShiftOperators{TSelf, TResult}.op_UnsignedRightShift(TSelf, int)" />
+        static short IShiftOperators<short, short>.operator >>>(short value, int shiftAmount) => (short)((ushort)value >>> shiftAmount);
```

Comment:
Why is the cast to ushort still needed here?

Reply:
C# doesn't define most operators for `byte`, `sbyte`, `short`, `ushort`, or `char` (outside the `op assignment` ones, like `+=` or `--`).

So in C#, `short >>> int` is actually `(int)short >>> int` which means `-1 >>> 1` goes from `0xFFFF` to `0xFFFF_FFFF` and then right shifts one giving `0x7FFF_FFFF`. Casting back to `short` then results in `0xFFFF` again.

Generic math returns and operates on the same type (like F# does) and so we need to explicitly treat it as a zero-extension so we get back `0x7FFF`